frontend/src/errors.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @file Custom Error classes. Useful as a way to see all potential errors that our system may throw/catch
+ */
+
+import { SearchFetcherProps } from "src/services/search/searchfetcher/SearchFetcher";
+
+/**
+ * A fetch request failed due to a network error. The error wasn't the fault of the user,
+ * and an issue was encountered while setting up or sending a request, or parsing the response.
+ * Examples of a NetworkError could be the user's device lost internet connection, a CORS issue,
+ * or a malformed request.
+ */
+
+export class NetworkError extends Error {
+  constructor(error: unknown, searchInputs: SearchFetcherProps) {
+    const serializedSearchInputs = convertSearchInputSetsToArrays(searchInputs);
+
+    const serializedData = JSON.stringify({
+      type: "NetworkError",
+      searchInputs: serializedSearchInputs,
+      message: error instanceof Error ? error.message : "Unknown Error",
+      status: 500,
+    });
+    super(serializedData);
+  }
+}
+
+// Used as a base class for all !response.ok errors
+export class BaseFrontendError extends Error {
+  constructor(
+    error: unknown,
+    searchInputs: SearchFetcherProps,
+    type: string,
+    status?: number,
+  ) {
+    const serializedSearchInputs = convertSearchInputSetsToArrays(searchInputs);
+
+    const serializedData = JSON.stringify({
+      type: type || "BaseFrontendError",
+      searchInputs: serializedSearchInputs,
+      message: error instanceof Error ? error.message : "Unknown Error",
+      status,
+    });
+
+    super(serializedData);
+
```

Comment:
stringify an object with error metadata so we can access on the error boundary component